File: analyze_m3.py
import pandas as pd
import json

file_path = 'C:/Users/DELL/Desktop/Campeche/Data_xls/Indicadores/anexo 3 (8 tablas) VF.xlsx'
xl = pd.ExcelFile(file_path)

summary = {}
for sheet_name in xl.sheet_names:
    if "ndice" in sheet_name.lower():
        continue
        
    df = xl.parse(sheet_name, header=None)
    
    # Get first 15 rows, non-empty cells
    rows_info = []
    for idx, row in df.head(15).iterrows():
        non_empty = []
        for col_idx, val in enumerate(row):
            if pd.notna(val) and str(val).strip() != "":
                col_letter = chr(65 + col_idx) if col_idx < 26 else chr(64 + col_idx//26) + chr(65 + col_idx%26)
                non_empty.append(f"{col_letter}: {val}")
        if non_empty:
            rows_info.append(f"Row {idx+1}: " + " | ".join(non_empty))
            
    summary[sheet_name] = rows_info

with open('m3_summary.txt', 'w', encoding='utf-8') as f:
    for sheet, rows in summary.items():
        f.write(f"=== {sheet} ===\n")
        for row in rows:
            f.write(f"{row}\n")
        f.write("\n")

File: analyze_m3.php
<?php
require 'vendor/autoload.php';
$filePath = 'C:/Users/DELL/Desktop/Campeche/Data_xls/Indicadores/anexo 3 (8 tablas) VF.xlsx';

if (!file_exists($filePath)) {
    echo "File not found: " . $filePath . "\n";
    exit(1);
}

try {
    $spreadsheet = \PhpOffice\PhpSpreadsheet\IOFactory::load($filePath);
    $sheets = $spreadsheet->getSheetNames();
    $output = [];
    foreach ($sheets as $sheetName) {
        $sheet = $spreadsheet->getSheetByName($sheetName);
        $data = $sheet->toArray(null, true, true, true);
        $output[$sheetName] = array_slice($data, 0, 20); // Top 20 rows
    }
    file_put_contents('m3_structure.json', json_encode($output, JSON_PRETTY_PRINT | JSON_UNESCAPED_UNICODE));
    echo "Done. Output saved to m3_structure.json\n";
} catch (Exception $e) {
    echo "Error: " . $e->getMessage() . "\n";
}
